console/backend/src/main/java/org/frankframework/console/filters/CorsFilter.java

```diff
@@ -117,4 +121,45 @@ public void destroy() {
 		// nothing to destroy
 	}
 
+	private static class OriginOverwritingHttpServletRequestWrapper extends HttpServletRequestWrapper {
+		private final boolean secure;
+
+		public OriginOverwritingHttpServletRequestWrapper(HttpServletRequest request) {
+			super(request);
+			secure = "https".equalsIgnoreCase(request.getScheme());
+		}
+
+		/**
+		 * If the endpoint is not secure (http) but the origin header is (mis-configured reverse proxy!)
+		 * Rewrite the only the `origin` header to HTTP.
+		 */
+		@Override
+		public String getHeader(String name) {
+			if(!secure && "origin".equalsIgnoreCase(name)) {
+				String originalValue = super.getHeader(name);
+				if(originalValue != null) {
+					return originalValue.replace("https:", "http:");
+				}
+			}
+			return super.getHeader(name);
+		}
+
+		@Override
+		public Enumeration<String> getHeaders(String name) {
+			if (!secure && "origin".equalsIgnoreCase(name)) {
+				Enumeration<String> originHeaders = super.getHeaders(name);
+				if (originHeaders != null) {
+					return Collections.enumeration(mapOriginHeaders(originHeaders));
+				}
+			}
+			return super.getHeaders(name);
+		}
+
+		private List<String> mapOriginHeaders(Enumeration<String> originHeaders) {
+			return Collections.list(originHeaders).stream()
+					.map(header -> header.replace("https:", "http:"))
```

Comment:
Je doet hier hetzelfde als op regel 141. Ik zou een simpele method hiervoor maken:
```
		private String replaceHttps(String originalValue) {
			return originalValue.replace("https:", "http:");
		}
```
en dan hier `this::replaceHttps`

Reply:
Ja achteraf, wil dit eerst werkend zien, mogelijk moet dit een losse class worden met tests ;)